### PR TITLE
[O2B-1319] Merge detectors tables  - remove DPL detectors table

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [ ] issue has "Fix version" assigned
- [ ] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Refactored DplDetector service to work upon Detectors

Changes made to the database:
- Removed dpl_detectors table

### CODE REVIEW

File: lib/server/controllers/detectors.controller.js [hunk 1]
```diff
@@ -23,7 +24,7 @@ const { updateExpressResponseFromNativeError } = require('../express/updateExpre
 exports.listAllDetectors = async (_request, response) => {
     try {
         const detectors = await detectorService.getAll();
-        response.status(200).json({ data: detectors });
+        response.status(200).json({ data: detectors.filter(({ type }) => [DetectorType.PHYSICAL, DetectorType.VIRTUAL].includes(type)) });
```
Comment (martinboulais) [inline lib/server/controllers/detectors.controller.js:27]: I suppose here you can use the constant?

**Review by martinboulais — DISMISSED** [review]

File: lib/public/services/detectors/detectorsProvider.js [hunk 2]
```diff
@@ -14,15 +14,15 @@
 import { Observable, RemoteData } from '/js/src/index.js';
 import { getRemoteData } from '../../utilities/fetch/getRemoteData.js';
 import { ObservableData } from '../../utilities/ObservableData.js';
-import { VIRTUAL_DETECTOR_NAME } from '../../domain/enums/detectorsNames.mjs';
+import { DATA_TAKING_DETECTOR_TYPES, DetectorType } from '../../domain/enums/DetectorTypes.mjs';
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/services/detectors/detectorsProvider.js]: yes, applied

File: lib/server/services/dpl/getOrCreateDplDetector.js [hunk 3]
```diff
@@ -12,11 +12,19 @@
...
  */
-exports.getOrCreateDplDetectorByName = (name) => DplDetectorRepository.findOneOrCreate({ name });
+exports.getOrCreateDplDetectorByName = async (name) => {
+    const dplDetector = await DplDetectorRepository.findOne({ where: { name } });
+    if (dplDetector) {
+        return dplDetector;
+    } else {
+        return await DplDetectorRepository.insert({ name, type: DetectorType.QC });
+    }
+};
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/server/services/dpl/getOrCreateDplDetector.js:30]: I stopped refactoring here on substituting `DplDetector` model for `Detector` one in `DplDetectorRepository`

File: lib/server/controllers/detectors.controller.js (hunk 1 above)
Comment (xsalonx) [reply lib/server/controllers/detectors.controller.js:27]: applied

File: lib/database/adapters/DetectorAdapter.js [hunk 4]
```diff
@@ -40,6 +40,22 @@ class DetectorAdapter {
         };
     }
 
+    /**
+     * Converts the given database object to a minified entity object.
+     *
+     * @param {SequelizeQcFlagType} databaseObject Object to convert.
```
Comment (graduta) [inline lib/database/adapters/DetectorAdapter.js:46]: Is the parameter type correct?

File: lib/database/adapters/QcFlagAdapter.js [hunk 5]
```diff
@@ -57,7 +57,7 @@ class QcFlagAdapter {
             comment: comment ?? null,
             origin,
             runNumber,
-            dplDetectorId,
+            dplDetectorId: detectorId,
```
Comment (graduta) [inline lib/database/adapters/QcFlagAdapter.js:60]: Is it on purpose left as `dplDefetctorId` in the adapter?

**Review by graduta — COMMENTED** [review]
It seems that we still use dpl naming in some places. Is this just the first out of many PRs to refactor? If so, are there tickets created documenting the remaining bits that need to be done?

File: lib/database/adapters/DetectorAdapter.js (hunk 4 above)
Comment (xsalonx) [reply lib/database/adapters/DetectorAdapter.js]: no, fixed

**Review by graduta — APPROVED** [review]

**Pull request merged**